### ISSUE TITLE
How can I avoid that a vehicle brakes hard when going straight downhill?

### ISSUE THREAD

**Issue by cantonsilver** [Stephan-S/FS22_AutoDrive#587]
Hello, everyone!
This is just a question not a bug report.
If I have a topographically demanding map like the "Waldstetten", how can I avoid that a vehicle brakes hard all the time when going straight downhill? This tends to occur when the descent is not smooth but has slight bumps. Turning off collision detection is not an option since I'm playing with AI traffic. I hope someone has a tip. Best regards, Andy

**Comment by 60erjonny** [comment]
![Screenshot (41)](https://user-images.githubusercontent.com/63659182/175805729-f1962742-9543-43ad-9c73-e6f31f36caca.png)
Kollisionshöhe über Grund mal versuchen zu ändern...

**Comment by cantonsilver** [comment]
> ![Screenshot (41)](https://user-images.githubusercontent.com/63659182/175805729-f1962742-9543-43ad-9c73-e6f31f36caca.png) Kollisionshöhe über Grund mal versuchen zu ändern...

Kollisionshöhe über Grund steht schon bei 2m. Da hab ich schon mit den verschiedensten Werten rum experimentiert. Das bringt es leider nicht. Eine kleine Verbesserung gibt es, wenn ich zusätzliche Punkte (und diese ultra eng) setze. Aber da wird man ja wahnsinnig..

**Comment by ThomasMatern** [comment]
@Axel32019 I've done a couple of tweaks to improve this (on dev_tommo). Two videos to illustrate:
old/current behaviour: https://drive.google.com/file/d/1Zw35_7NsGhOT0smpoaJl71wWlkVjzg0W/view?usp=drive_link
improved behaviour: https://drive.google.com/file/d/1vVpH4xIZr3DBsnIJGWw4YoANJcuYYrsO/view?usp=drive_link

These are just tweaks to the existing collision detection. Using a single hitbox to detect collisions for a vehicle following a path will always be crude. Ideally, AD would use multiple hitboxes and align them with the waypoints. However, this would be a major change and probably not worth the effort.

This will need a bit of testing to ensure I didn't break something else.

**Comment by regs1980** [comment]
@ThomasMatern Kann man wirklich so einbauen in den AD, ist eine sehr gute Option.

Ist auf jeden Fall eine Verbesserung bei Karten mit extremen Höhenunterschieden in Verbindung mit Schranken usw. 
Kollisionssensor alt
https://drive.google.com/file/d/1S4MA9AIBPDCP9avU7pshvb98SHaq4N0V/view?usp=sharing
Kollisionssensor neu
https://drive.google.com/file/d/1aRBkg9b97_9ifwnfg8n5cZ0V-ymvWp24/view?usp=sharing

Der einzige Nachteil ist Gegenstände, die kleiner sind als der eingestellte Wert nicht mehr erkannt werden.
Die alte Variante hatte halt noch ein kleines Offset nach unten.
Schaut man sich die Videos an, ist der Nachteil völlig zu verschmerzen, bei einer Kollisionshöhe von einem Meter erkennt er keinen Baumstamm oder 120er-Ballen mehr aber auf jeden Fall eine vorausfahren PKW in der Steillage im Video.

**Comment by Axel32019** [comment]
@ThomasMatern 
So much difference I do not see between the 2 videos.
'Ideally, AD would use multiple hitboxes and align them with the waypoints.'
This would create much effort to implement and will have a considerable impact on performance, as this will run in each update.
An idea could be to add some debounce to the colli detection, i.e. 0.2...0.5 s.

To solve #211 I have the following ideas related to this topic:
- reduce the minimum colli detection to 0.1...0.5 m, at the moment it is 2 m
- add an additional sensor to detect only vehicles with < 2 m distance, which is needed to ensure a sufficient distance to forward vehicle when opening the back door

**Comment by ThomasMatern** [comment]
@Axel32019 thanks for looking into this. I went frame-counting my videos for you. Both use the same starting point and vehicles. I checked the section from the point where the tractor hits 1km/h to the first blue water barrel:
Old: 56s, tractor brakes for 309 frames, it slows down to 12km/h twice and down to 9km/h once
New: 43s, tractor brakes for 102 frames, it slows down to 35km/h three times and 26km/h once. 
I would call this a significant improvement.
The videos @regs1980 recorded, show a significant improvement in the area where the truck drives down a long slope.

I looked into the debounce initially, but it causes problems when there is an actual reason to stop, e.g. another vehicle or a bale. @regs1980 second video illustrates this right at the start. Delaying the braking for 0.2s would cause the tractor to hit the bale.
(side note: it's quite funny to see the white NPC car braking hard for no reason just before hitting the sloped section at the end... seems Giants has similar issues)

Re. #211 sounds like a good idea, I'll give this a go
